openstack/objectstorage/v1/objects/requests.go

```diff
@@ -456,6 +460,24 @@ type CreateTempURLOpts struct {
 	Split string
 }
 
+type nowFuncT func() time.Time
```

Comment:
What if add the date into the `CreateTempURLOpts` struct? If it is nil, then a default `time.Now()` is used. If not, the provided value is used. This will avoid hacking around the `ResetClockImplementation`

Reply:
See here for example: https://github.com/gophercloud/gophercloud/blob/v0.12.0/openstack/identity/v3/extensions/ec2tokens/requests.go#L74
https://github.com/gophercloud/gophercloud/blob/v0.12.0/openstack/identity/v3/extensions/ec2tokens/requests.go#L268

Reply:
Sounds good to me

Reply:
Please check 9a500de9eaad35ab3e640fe323ccfe47bf88532d

Notice I didn't use pointer to time.Time because as documentation says:
> Programs using times should typically store and pass them as values, not pointers. That is, time variables and struct fields should be of type time.Time, not *time.Time. 

But I used `isZero()` to check for null value.

https://golang.org/pkg/time/#Time

Reply:
I looks much cleaner :-)